### DISCUSSION TITLE
parse_key_value: howto discard stand-alone keys

### DISCUSSION THREAD

**Discussion by tshepang** [vectordotdev/vector#13419] [General]
```vrl
$ parse_key_value!("one=1 two=2 three", accept_standalone_key: false)
function call error for "parse_key_value" at (0:67): could not parse whole line successfully
```
Would be nice to have an option to simply discard `three` and keep the rest. Is there a workaround?

**Comment by jszwedko** [comment]
Hi @tshepang ,

One way to do this would be to use `accept_standalone_key: true` and then filter the event afterwards to drop any keys that are set to `true`, though this would result in a false positive if there was a `three=true` in the string. If you want to open a feature request, we can see about updating `parse_key_value` to have additional options to ignore non-key/value parts.

**Reply by tshepang** [reply]
That option gives...

> function call error for "parse_key_value"...

Would be nice indeed to have additional options for this function.

**Reply by tshepang** [reply]
I created #13447
